### ISSUE TITLE
[Feature Request]: Standardize breadcrumbs throughout the site

### ISSUE THREAD

**Issue by S4v8n** [oppia/design-team#216]
### Is your feature request related to a problem? Please describe.

I am confused when I navigate through Oppia's Math page and Community lessons.  When I want to learn other topics in the subject, I don't have access to the hierachical subject page when I am in a lesson.  This often forces me to start over from Oppia's front page or the drop down menu.    

### Describe the solution (or solutions) you'd like

The solution is to add a breadcrumbs feature. This would allow the user to go to the corresponding subject page related to the lesson.  This would alleviate any frustration / confusion that mounts while navigating through Oppia's site.  I used Math page, Community Library page, and Search result page as example pages to add the feature but note that not all pages will need breadcrumbs.  Please refer to the [figma file](https://www.figma.com/design/xxVpFup4bbUeB2Y5jNv70t/Oppia-%2319918-Bread-Crumbs?node-id=142-40822&t=9s03pvysJNRc0nux-1), for more detailed instructions on pages needing breadcrumbs read "Breadcrumbs logic 2.0".

Currently on Oppia's site there are evidences of breadcrumbs in the Donate, Volunteer, and About page etc.  All existing breadcrumbs will be reformatted to the new version.  An audit of the entire site will be needed to update or remove existing breadcrumbs.      

The plan for breadcrumbs is go three levels deep maximum.  Going past three levels will take up too much space and also become quite busy.  Also to manage the length of the bread crumbs and visual clarity, I would like to set the font size to 14pt, placed below the nav bar and aligned with the title heading text.

See example image for reference.   

<img width="752" alt="image" src="https://github.com/oppia/design-team/assets/140122208/233fe328-1f82-4d44-b944-c39afca1cd84">

### Describe alternatives you've considered and rejected

I have considered going to four levels deep and possibly five levels. With the lengthy lesson titles and also subject names, I believe simplicity wins by keeping a rule of three levels.  

### Additional context

Here is how I envision the breadcrumbs to display on the pages.  

From Math Page

Oppia > Math - User sees this at the math classroom page
Oppia > Math > Topic Name - User sees this after selecting a topic
Oppia > Math > Topic Name - User sees this at chapter lesson page and each lesson player page
Oppia > Library -  User sees this in search results page after searching from search bar on Math Page

From Community Library Page:

 Oppia > Library
Oppia > Library > Subject name - The user will see after selecting the subject, the collection name, and the non collection name lesson.  This way the user can backtrack to the same subject for other lessons. 

From Oppia Learner members:

No display - Homepage or Learner’s Dashboard page for logged in users.  After log in, user will not see breadcrumbs at the dashboard page
Oppia > Math > Topic Name - User sees this after selecting lessons to resume or try new lessons
Oppia > Library - User sees this after searching from learner dashboard
Oppia > Library > Subject name - User sees this after selecting a community lesson from the dashboard.

Lesson players and Practice lessons:

No breadcrumbs will be needed. Only display the Lesson number and name below the nav bar. 

Pages that do not need breadcrumbs: 

Oppia Homepage
About page
About Oppia foundation page
Blog page
Contact page
Learner Dashboard page
Partnerships page
Volunteers page
Creator Dashboard
Contributor Dashboard
Donate page
Preference page

[Mobile] Breadcrumbs rule: use no more than 1 Level deep for breadcrumbs.

Should have a “<” followed by “Back to...” corresponding hierarchal page.
Follows the same rule for pages needing breadcrumbs.

**Comment by seanlip** [comment]
@S4v8n Thanks for filing this issue. Could you please specify a few things:

- What happens to the breadcrumbs as the width of the page narrows -- what does the full responsive behaviour look like?
- Should it be possible to click on each item in the breadcrumb? If so, where does each click take the user?

Thanks!

**Comment by seanlip** [comment]
Also one note ... TBH the text on the left feels a bit disjointed to me since "Oppia" is so much larger than the stuff next to it. This makes it feel a bit jarring IMO, partly also because of the extra space on the left of the ">". Do you feel that this is the case too, and might it be worth getting more design feedback on it from the team? (Of course, I might be the outlier here too, and it's fine to ignore this note if you like.)

**Comment by S4v8n** [comment]
@seanlip I was looking to make the Oppia logo more prominent and the secondary would be the breadcrumbs with the reduced font size (14pt).  The font size also aligns with the same size as the navigation bar menu.  I have considered the font to be a little larger  (16pt) but that looks like it was unintentional.  

I do apologize for the not adjusting the spacing between the Oppia and the arrows and the 'Math' text.  Thanks for pointing that out.  Let me play with the spacing for a bit.   Also, I do plan to run it by the team in our weekly meeting to gather their feedback as well.  


Regarding the responsive question: 

There are three options:  
- a. behavior follows how it is currently, as soon as the screen size reduces navigation menu titles will move closer to the breadcrumbs, once it is in contact with the breadcrumbs, the bread crumbs disappears.  For mobile phone screens the breadcrumbs will not appear. 
- b. breadcrumb will reduce in length by having a '...' to replace the second level text and only display the third level breadcrumbs.  once the user goes to the second level, the bread crumbs will display arrow and text. 
- c. breadcrumb text is responsive and reduces according to screen size.  we would need to set limits on how small the fonts should be.  The challenge here is how accessible it is to the user if the text gets too small.  


Regarding question about what happens when user clicks on the breadcrumbs. 
yes, the use will be able to click on the breadcrumbs. 
For example: if the user in a lesson and the breadcrumbs display ' Oppia > Math > Place Values' 
If the clicks on the third level (Place values), it will take user back to the Place Values page to view all other lessons.  if the user clicks on the second level (Math) it will take user back to the math page to view all topics. 


What are your thoughts?

**Comment by seanlip** [comment]
Hi @S4v8n -- sorry for the late reply. I don't have a strong opinion yet on what should happen here, I suggest letting the conversation with the design team play out so we can finalize the spec. For now, I'll put a note to say this issue is blocked on the design team, and we can remove it once there's a consensus reached. Thanks!

**Comment by S4v8n** [comment]
Hi @Sean Lip ***@***.***>,

No problem, I have created some alternate options to review/ gather
feedback with the team.  Thanks for the update.

**Comment by S4v8n** [comment]
Hi team, 

My name is Kurin Tu, and I am the UX designer heading up the project for the creating a standard for breadcrumbs.  The standardization of breadcrumbs across the entire site has been finalized.  Please review this [figma file ](https://www.figma.com/design/xxVpFup4bbUeB2Y5jNv70t/Oppia-%2319918-Bread-Crumbs?node-id=142-40822&t=9Bhzf5jYJEfhNwSt-1)for more details.   Feel free contact me with any questions.   Thank you!

**Comment by DevanshDeep-5** [comment]
Hi @S4v8n , @seanlip 
I am interested in solving this issue, but I have doubt that I have to code everything that is there in that Figma file.

**Comment by seanlip** [comment]
@DevanshDeep-5 Yes, please tackle the full issue and try to keep the code for breadcrumbs consistent across all pages. Try to think about this as implementing a single breadcrumb "component" and using it in multiple places, rather than separate implementations for each page.

**Comment by cmorocho11** [comment]
@seanlip would I be able to work on this issue? It seems do able but might need some help/feedback if thats okay.

**Comment by seanlip** [comment]
@cmorocho11 Per the guidance at https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#finding-something-to-do, please provide an explanation of what your PR will do (with names of files you're changing, what you plan to change in each file, etc.), as well as a video showing that the changes work correctly on your local machine. If it looks good, we can assign you to this issue. If you think that that might be difficult to do directly, I suggest working on a different issue instead.

Please also follow the other instructions on that wiki page if you have not yet done so. Thanks!

**Comment by HardikGoyal2003** [comment]
Hey @S4v8n,

I have few questions:
- Are `About page` and `About Oppia foundation page` same?
- Why are we removing the breadcrumb from the blog page? Current design looks good? Similar design we have on the /blog-dashboard page, so we have change over there also?

![Image](https://github.com/user-attachments/assets/0b770a24-e96d-4e4c-baa9-896b01067444)

![Image](https://github.com/user-attachments/assets/c1895362-4649-4712-86cb-856381ab1b41)

- There are many pages not mentioned in the above issue, so should I assume that we will be removing the breadcrumbs from all of them?

**Comment by S4v8n** [comment]
Hi @HardikGoyal2003 

- Yes, about page and about Oppia are the same page
- Primarily, it is for consistency and clarity when it comes to wayfinding and branding. The breadcrumbs are designed to help the user to be able to wayfind back to the previous screens when they are 2 to 3 screens deep into a topic.  The pages like the blog page, about, volunteer, donate, etc are all single level pages which does not require breadcrumbs.  I have presented this in the description above and also in the 'breadcrumbs logic 2.0' of the figma file,  please review them if you haven't already.   Secondly, having the wayfinding feature integrated with the Oppia brand does not provide clear distinction of the Oppia logo, the design team thought it would be best to separate out the wayfinding feature from the logo.   Lastly, by using the breadcrumbs feature in the new location, alleviates the spacing issues we are currently facing with long titles which inteferes with the navigation menu titles. 
- "There are many pages not mentioned in the above issue, so should I assume that we will be removing the breadcrumbs from all of them?"   yes, please remove the existing wayfinding method because it is not needed. The breadcrumbs should be implemented as part of the classroom subjects and topics.

**Comment by HardikGoyal2003** [comment]
@S4v8n I have a few more questions:

- How will we navigate back to the /blog page from a certain blog, i.e, /blog/<blog_id>?
- There are no mocks for some pages, but I think we need to keep the breadcrumb for some pages. Can you please review those?
  - Topic Editor Page
![Image](https://github.com/user-attachments/assets/7af43473-9941-40c5-874c-e10d3e222cf7)

  - Story Editor Page
![Image](https://github.com/user-attachments/assets/f71b6ebd-3739-4dab-a29c-5cb7033ee1b8)
  
  - Chapter Editor Page
![Image](https://github.com/user-attachments/assets/e96f3c9b-4616-4536-aabc-003394c2df02)

  - Story Viewer Page
![Image](https://github.com/user-attachments/assets/de6a1598-0f38-4a4b-96d6-5386dc4f56af)
  
  - Skill Editor Page
![Image](https://github.com/user-attachments/assets/dc459ab1-282b-4f68-8cfc-e7ad00fa55db)

  - Profile Page (For this page, I don't think it's needed)
![Image](https://github.com/user-attachments/assets/b58e8017-a22f-4eb6-b49e-730641ae1c2a)

   - Collection Editor Page
![Image](https://github.com/user-attachments/assets/5e12cd68-b2d8-4e29-a269-375f2f2de253)

  - Collection Viewer Page
![Image](https://github.com/user-attachments/assets/09de3a8e-3bb5-4b2f-9fa4-6921ff04076d)

  - Exploration Editor Page
![Image](https://github.com/user-attachments/assets/070ffc6c-ea0b-4e6f-a043-e8c2a99472b4)

**Comment by S4v8n** [comment]
Hi @HardikGoyal2003 
For the blog page, once the user goes to an article, I would recommend using a "< Back to Blog" wayfinding. See image below for reference.  The main blog page does not a need a breadcrumbs feature only because it does not go past 2 levels.    Also note when the user clicks on the "< Back to Blog" wayfinding feature, it should take the user back to the most recent page of the blog.  For example if the user is scrolling through and selects an article in page 2 of the blog, after the user is finished reading the article and clicks "< Back to Blog" it should take the user to page 2 of the blog. From there the user can decide to go to whichever page they want.    

<img width="970" alt="Image" src="https://github.com/user-attachments/assets/85c50ea0-b4ff-484b-89f0-7003c6c353c7" />

As for the remaining pages that you brought up. Thank you for bringing it to my attention.  I don't think I have access to these screens nor was I made aware of them.  I will need some time to go through them and see the interactions.  @seanlip Is there a way for me to access these pages?  I do have access to the creator and contributor dashboard, I was able to see the interactions between the creating a lesson and translations but did not get to see the topics/skills/ story pages.  

For the profile page, agreed breadcrumbs feature is not needing.

**Comment by seanlip** [comment]
(Just a note, I've given @S4v8n the necessary access to the requested pages.)

**Comment by HardikGoyal2003** [comment]
Hey @S4v8n can you please specify a date when mocks will be ready?

**Comment by S4v8n** [comment]
Hi Hardik,

I will work on getting them over to you by the end of the week.

Best,
Kurin

**Comment by HardikGoyal2003** [comment]
@S4v8n Any updates on the designs for this issue?

**Comment by S4v8n** [comment]
Hi Hardik,

Sorry for the delay, I'm still working on getting the frames completed.
Will update you by the end of this week. Thank you for your patience.

**Comment by HardikGoyal2003** [comment]
Hey @S4v8n Any updates on this?

**Comment by S4v8n** [comment]
Hi @HardikGoyal2003 , I was sidetracked with work and is back to working on it this week.  This turned out to a much bigger project in itself as it relates to everything on the contributor side.  I will try to get it completed as soon as I can.

**Comment by tanvishinde017** [comment]
Hi @HardikGoyal2003, @S4v8n, 
I’m a new contributor preparing for GSoC. 
I would like to work on implementing the breadcrumbs for the Math page as per the specs. 
Could you please assign me this issue?

**Comment by HardikGoyal2003** [comment]
Hi @tanvishinde017 This issue is blocked on design currently. You can pick up another issue from here: https://github.com/oppia/oppia/issues?q=state%3Aopen%20label%3A%22good%20first%20issue%22. Thanks!

**Comment by sravanya1912** [comment]
@HardikGoyal2003 @S4v8n  
 Hi, I’ve gone through the issue ,  if it is still open i would like to work on it.

**Comment by HardikGoyal2003** [comment]
Hi @sravanya1912 This issue is blocked on design currently. You can pick up another issue from here: https://github.com/oppia/oppia/issues?q=state%3Aopen%20label%3A%22good%20first%20issue%22. Thanks!
